util/chunk/iterator_test.go

```diff
@@ -43,16 +43,55 @@ func (s *testChunkSuite) TestIterator(c *check.C) {
 
 	it := NewIterator4Slice(rows)
 	checkIterator(c, it, expected)
+	for i := 0; i > 5; i++ {
+		c.Assert(it.Current(), check.Equals, rows[i])
+		it.Next()
+	}
+	it.ReachEnd()
+	c.Assert(it.Current(), check.Equals, it.End())
+	c.Assert(it.Begin(), check.Equals, rows[0])
+
 	it = NewIterator4Chunk(chk)
 	checkIterator(c, it, expected)
+	for i := 0; i > 5; i++ {
+		c.Assert(it.Current(), check.Equals, chk.GetRow(i))
+		it.Next()
+	}
+	it.ReachEnd()
+	c.Assert(it.Current(), check.Equals, it.End())
+	c.Assert(it.Begin(), check.Equals, chk.GetRow(0))
+
 	it = NewIterator4List(li)
 	checkIterator(c, it, expected)
+	checkIterator(c, it, expected)
```

Comment:
Check it twice?

Reply:
The loop can not be extracted as a function,
since the last parameter of c.Assert() is not the same type.